public/components/integrations/components/setup_integration.tsx

```diff
@@ -318,75 +220,69 @@ export function SetupBottomBar({
               window.location.hash = hash;
             }}
           >
-            Cancel
-          </Eui.EuiButton>
-        </Eui.EuiFlexItem>
-        <Eui.EuiFlexItem>
-          <Eui.EuiSpacer />
-        </Eui.EuiFlexItem>
-        {step > 0 ? (
-          <Eui.EuiFlexItem grow={false}>
-            <Eui.EuiButton iconType={'returnKey'} onClick={() => setStep(step - 1)}>
-              Back
-            </Eui.EuiButton>
-          </Eui.EuiFlexItem>
-        ) : null}
-        <Eui.EuiFlexItem grow={false}>
-          <Eui.EuiButton
+            Discard
+          </EuiButton>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButton
             fill
-            iconType={'check'}
-            onClick={() => {
-              if (step < STEPS.length - 1) {
-                setStep(step + 1);
-              } else {
-                console.log(config);
-              }
+            iconType="arrowRight"
+            iconSide="right"
+            isLoading={loading}
+            onClick={async () => {
+              setLoading(true);
+              const template = await findTemplate(integration.name);
+              await addIntegrationRequest(
+                false,
+                integration.name,
+                config.displayName,
+                template,
+                setToast,
+                config.displayName,
+                config.connectionDataSource
+              );
+              setLoading(false);
             }}
           >
-            {step === STEPS.length - 1 ? 'Save' : 'Next'}
-          </Eui.EuiButton>
-        </Eui.EuiFlexItem>
-      </Eui.EuiFlexGroup>
-    </Eui.EuiBottomBar>
+            Add Integration
+          </EuiButton>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiBottomBar>
   );
 }
 
-export function SetupIntegrationStepsPage() {
+export function SetupIntegrationPage({
+  integration,
+}: {
+  integration: {
+    name: string;
+    type: string;
+  };
+}) {
   const [integConfig, setConfig] = useState({
-    instanceName: '',
-    useExisting: true,
-    dataSourceName: '',
-    dataSourceDescription: '',
-    dataSourceFileType: 'parquet',
-    dataSourceLocation: '',
-    existingDataSourceName: '',
+    displayName: `${integration.name} Integration`,
+    connectionType: 'index',
+    connectionDataSource: '',
   } as IntegrationConfig);
```

Comment:
curious: Why is this type declared at two places: https://github.com/Swiddis/dashboards-observability/blob/d67ec95620df79b8e1b8e0cb7d6528afd7c7758c/server/adaptors/integrations/types.ts#L8 and https://github.com/Swiddis/dashboards-observability/blob/d67ec95620df79b8e1b8e0cb7d6528afd7c7758c/public/components/integrations/components/setup_integration.tsx#L10. 

Also, if possible we should try to initialize all fields so that you don't need to use `{} as IntegrationConfig`.